### PR TITLE
Disallow multiple assignment for constants. Fixes #3466

### LINKED ISSUE #3466: Multiple constant assignment regression
``` crystal
FOO, BAR = 1, 2
FOO # => undefined constant FOO
```

This was fixed after being reported reported in #958 but it's failing again with a different error message.

### PR DESCRIPTION
**Pull request by sdogruyol**
This is a reopen of https://github.com/crystal-lang/crystal/pull/3507. #3507 got messed up and couldn't find a way to fix `unknown repository` error on Github.

### CODE REVIEW

File: spec/compiler/parser/parser_spec.cr [hunk 1]
```diff
@@ -1324,6 +1323,8 @@ describe "Parser" do
     "can't change the value of self"
   assert_syntax_error "self += 1",
     "can't change the value of self"
+  assert_syntax_error "FOO, BAR = 1, 2",
```
Comment (sdogruyol) [inline spec/compiler/parser/parser_spec.cr]: @asterite IDK why but this fails with

```
  1) Parser says syntax error on "FOO, BAR = 1, 2"
     Failure/Error: assert_syntax_error "FOO, BAR = 1, 2",

       expected message to include "Multiple assignment is not allowed for constants" but got "unexpected token: ,"
```

Comment (sdogruyol) [inline spec/compiler/parser/parser_spec.cr]: I've fixed it 🎉

**Comment by sdogruyol** [comment]
@asterite i've updated the PR

**Comment by asterite** [comment]
@sdogruyol Thank you! 🎉

**Pull request merged**